packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -0,0 +1,116 @@
+};
+
+const getStoredList = (resourceName: string): { cloudFormationParameterName: string; environmaneVariableName: string }[] => {
+  const projectBackendDirPath = pathManager.getBackendDirPath();
+  const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const functionParameterFilePath = path.join(resourcePath, functionParametersFileName);
+  const functionParameters = (JSONUtilities.readJson(functionParameterFilePath, { throwIfNotExist: false }) as object) || {};
+  return _.get(functionParameters, 'environmentVariableList', []);
+};
+
+const updateStoredList = (resourceName: string, newList: object): void => {
+  const projectBackendDirPath = pathManager.getBackendDirPath();
+  const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const functionParameterFilePath = path.join(resourcePath, functionParametersFileName);
+  const functionParameters = (JSONUtilities.readJson(functionParameterFilePath, { throwIfNotExist: false }) as object) || {};
+  _.set(functionParameters, 'environmentVariableList', newList);
+  JSONUtilities.writeJson(functionParameterFilePath, functionParameters);
+};
+
+const getStoredReferences = (resourceName: string): object => {
+  const projectBackendDirPath = pathManager.getBackendDirPath();
+  const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const cfnFileName = `${resourceName}-cloudformation-template.json`;
+  const cfnFilePath = path.join(resourcePath, cfnFileName);
+  const cfnContent = JSONUtilities.readJson(cfnFilePath, { throwIfNotExist: false }) || {};
+  return _.get(cfnContent, ['Resources', 'LambdaFunction', 'Properties', 'Environment', 'Variables'], {});
+};
+
+const updateStoredReferences = (resourceName: string, newReferences: object): void => {
+  const projectBackendDirPath = pathManager.getBackendDirPath();
+  const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const cfnFileName = `${resourceName}-cloudformation-template.json`;
+  const cfnFilePath = path.join(resourcePath, cfnFileName);
+  const cfnContent = (JSONUtilities.readJson(cfnFilePath, { throwIfNotExist: false }) as object) || {};
+  _.set(cfnContent, ['Resources', 'LambdaFunction', 'Properties', 'Environment', 'Variables'], newReferences);
+  JSONUtilities.writeJson(cfnFilePath, cfnContent);
+};
+
+const getStoredKeyValue = (context: $TSContext, resourceName: string): object => {
+  const teamProviderInfoPath = pathManager.getTeamProviderInfoFilePath();
+  const { envName } = context.amplify.getEnvInfo();
+  const teamProviderInfo = JSONUtilities.readJson(teamProviderInfoPath, { throwIfNotExist: false }) || {};
+  return _.get(teamProviderInfo, [envName, 'categories', categoryName, resourceName], {});
+};
+
+const updateStoredKeyValue = (context: $TSContext, resourceName: string, newKeyValue: object): void => {
+  const teamProviderInfoPath = pathManager.getTeamProviderInfoFilePath();
```

Comment:
use `stateManager.getTeamProviderInfo()` and `stateManager.setTeamProviderInfo()`

Reply:
Sounds good. Updated!